kotlin/workflow-ui-android/src/main/java/com/squareup/workflow/ui/ViewShowRendering.kt

```diff
@@ -58,30 +58,60 @@ fun <RenderingT : Any> View.bindShowRendering(
  * [rendering] and the new one.
  */
 fun View.canShowRendering(rendering: Any): Boolean {
-  return showRenderingTag?.initialRendering?.matches(rendering) == true
+  return getRendering<Any>()?.matches(rendering) == true
 }
 
 /**
- * Shows [rendering] in a view that has been initialized by [bindShowRendering].
+ * Sets the workflow rendering associated with this view, and displays it by
+ * invoking the [ViewShowRendering] function previously set by [bindShowRendering].
+ *
+ * @throws IllegalStateException if [bindShowRendering] has not been called.
  */
 fun <RenderingT : Any> View.showRendering(rendering: RenderingT) {
   showRenderingTag
       ?.let { tag ->
-        check(tag.initialRendering.matches(rendering)) {
-          "Expected $this to be able show $rendering, should have matched ${tag.initialRendering}."
+        check(tag.showing.matches(rendering)) {
+          "Expected $this to be able show $rendering, should have matched ${tag.showing}. " +
+              "Consider using ${WorkflowViewStub::class.java.simpleName} to display arbitrary types."
         }
 
-        @Suppress("UNCHECKED_CAST")
-        (tag.showRendering as ViewShowRendering<RenderingT>).invoke(rendering)
+        bindShowRendering(rendering, tag.showRendering)
       }
       ?: error("Expected $this to have a showRendering function for $rendering.")
 }
 
+/**
+ * Returns the most recent rendering shown by this view, or null if [bindShowRendering]
+ * has never been called.
+ *
+ * @throws ClassCastException if a rendering is found but is not of type [RenderingT]
+ */
+inline fun <reified RenderingT : Any> View.getRendering(): RenderingT? {
+  return when (val showing = showRenderingTag?.showing) {
+    null -> null
+    else -> showing as RenderingT
```

Comment:
Is this function inline just for this cast? What happens if you just cast without it being inline?

Reply:
I _think_ it won't compile. I'll check again. If it's just a warning, I'll try to undo some of the inlining.

Reply:
Okay, it's just a warning. Sanity check please: that should in theory be okay since a real cast will happen as soon as the return value is assigned, right?

Reply:
Yes, I believe the cast error will come from the assignment.